lib/galaxy/webapps/galaxy/api/library_contents.py

```diff
@@ -3,506 +3,169 @@
-        return rval
+router = Router(tags=["libraries"])
+
+
+class FormDataApiRoute(APIContentTypeRoute):
+    match_content_type = "multipart/form-data"
+
+
+class JsonApiRoute(APIContentTypeRoute):
+    match_content_type = "application/json"
+
+
+LibraryContentsCreateForm = as_form(LibraryContentsFileCreatePayload)
+
+
+@router.cbv
+class FastAPILibraryContents:
+    service: LibraryContentsService = depends(LibraryContentsService)
+
+    @router.get(
+        "/api/libraries/{library_id}/contents",
+        summary="Return a list of library files and folders.",
+        deprecated=True,
+    )
+    def index(
+        self,
+        library_id: DecodedDatabaseIdField,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> LibraryContentsIndexListResponse:
+        """This endpoint is deprecated. Please use GET /api/folders/{folder_id}/contents instead."""
+        return self.service.index(trans, library_id)
+
+    @router.get(
+        "/api/libraries/{library_id}/contents/{id}",
+        name="library_content",
+        summary="Return a library file or folder.",
+        deprecated=True,
+    )
+    def show(
+        self,
+        library_id: DecodedDatabaseIdField,
+        id: MaybeLibraryFolderOrDatasetID,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> AnyLibraryContentsShowResponse:
+        """This endpoint is deprecated. Please use GET /api/libraries/datasets/{library_id} instead."""
+        return self.service.show(trans, id)
+
+    @router.post(
+        "/api/libraries/{library_id}/contents",
+        summary="Create a new library file or folder.",
+        deprecated=True,
+        route_class_override=JsonApiRoute,
+    )
+    def create_json(
+        self,
+        library_id: DecodedDatabaseIdField,
+        payload: AnyLibraryContentsCreatePayload,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+    ) -> AnyLibraryContentsCreateResponse:
+        """This endpoint is deprecated. Please use POST /api/folders/{folder_id} or POST /api/folders/{folder_id}/contents instead."""
+        return self.service.create(trans, library_id, payload)
+
+    @router.post(
+        "/api/libraries/{library_id}/contents",
+        summary="Create a new library file or folder.",
+        deprecated=True,
+        route_class_override=FormDataApiRoute,
+    )
+    async def create_form(
+        self,
+        request: Request,
+        library_id: DecodedDatabaseIdField,
+        payload: LibraryContentsFileCreatePayload = Depends(LibraryContentsCreateForm.as_form),
+        files: Optional[List[UploadFile]] = None,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+    ) -> AnyLibraryContentsCreateResponse:
+        """This endpoint is deprecated. Please use POST /api/folders/{folder_id} or POST /api/folders/{folder_id}/contents instead."""
+        # FastAPI's UploadFile is a very light wrapper around starlette's UploadFile
+        if not files:
```

Comment:
You're doing I/O here with sync libraries, can you take a look at https://github.com/galaxyproject/galaxy/pull/18944 and move the async part into a dependency ?